services/ymax-planner/src/engine.ts

```diff
@@ -192,6 +193,57 @@ type ProcessPortfolioPowers = Pick<
   };
 };
 
+/**
+ * Determines if a plan needs to be submitted for a given portfolio flow.
+ * A plan is needed if:
+ * - The flow does not have vstorage data (flowKeys does not contain flowKey), OR
+ * - The flow exists but has status: 'init'
+ *
+ * @param portfolioKey - The portfolio key (e.g., "portfolio1")
+ * @param flowKey - The flow key (e.g., "flow1")
+ * @param flowKeys - Set of flow keys that have vstorage data
+ * @param vstorage - Vstorage query interface
+ * @param portfoliosPathPrefix - Prefix for portfolio vstorage paths
+ * @param marshaller - Marshaller for reading vstorage data
+ * @param readOpts - Options for reading vstorage
+ * @param readOpts.minBlockHeight - Minimum block height for reading
+ * @param readOpts.retries - Number of retries for reading
+ * @returns Promise<boolean> - true if a plan needs to be submitted
+ */
+export const planNeeded = async (
+  portfolioKey: string,
```

Comment:
Use static types, not examples, to document syntax of such args:

```suggestion
  portfolioKey: `portfolio${number}`,
```

likewise for `flowKey`, `flowKeys`